### ISSUE TITLE
QT 5.5.1 Failing to Build on Mac OS X 10.7.5

### ISSUE THREAD

**Issue by dagoodma** [Homebrew/homebrew#45284]
Hello,

I am trying to upgrade Qt 5.3.2 on Mac OS X 10.7.5 to Qt 5.5.1. The build fails after a few hours in. Here's the output:

    $ brew install qt5 --with-developer
    ==> Downloading https://download.qt.io/official_releases/qt/5.5/5.5.1/single/qt-everywhere-opensource-src-5.5.1.tar.xz
    Already downloaded: /Library/Caches/Homebrew/qt5-5.5.1.tar.xz
    ==> Downloading https://raw.githubusercontent.com/Homebrew/patches/2fcc1f8ec1df1c90785f4fa6632cebac68772fa9/qt5/el-capitan-2.dif
    Already downloaded: /Library/Caches/Homebrew/qt5--patch-b8f04efd047eeed7cfd15b029ece20b5fe3c0960b74f7a5cb98bd36475463227.diff
    ==> Downloading https://gist.githubusercontent.com/UniqMartin/a54542d666be1983dc83/raw/f235dfb418c3d0d086c3baae520d538bae0b1c70/
    Already downloaded: /Library/Caches/Homebrew/qt5--patch-e31df5d0c5f8a9e738823299cb6ed5f5951314a28d4a4f9f021f423963038432.patch
    ==> Patching
    patching file qtwebengine/src/3rdparty/chromium/chrome/installer/mac/third_party/bsdiff/goobsdiff.gyp
    patching file qtwebengine/src/3rdparty/chromium/content/content_browser.gypi
    patching file qtwebengine/src/3rdparty/chromium/net/net.gyp
    patching file qtwebengine/src/3rdparty/chromium/printing/printing.gyp
    patching file qtwebengine/src/3rdparty/chromium/sandbox/mac/sandbox_mac.gypi
    patching file qtwebengine/src/3rdparty/chromium/third_party/ffmpeg/ffmpeg.gyp
    patching file qtwebengine/src/3rdparty/chromium/third_party/libxml/libxml.gyp
    patching file qtwebengine/src/3rdparty/chromium/third_party/skia/gyp/zlib.gyp
    patching file qtwebengine/src/3rdparty/chromium/third_party/sqlite/sqlite.gyp
    patching file qtwebengine/src/3rdparty/chromium/tools/gyp/pylib/gyp/generator/xcode.py
    patch unexpectedly ends in middle of line
    Hunk #1 succeeded at 88 with fuzz 2.
    patching file qtbase/qmake/generators/makefile.cpp
    ==> ./configure -prefix /usr/local/Cellar/qt5/5.5.1 -release -opensource -confirm-license -system-zlib -qt-libpng -qt-libjpeg -n
    ==> make
    Last 15 lines from /Users/dagoodma/Library/Logs/Homebrew/qt5/02.make:
    inline _LIBCPP_INLINE_VISIBILITY const char* strchr(const char* __s, int __c) {return ::strchr(__s, __c);}
                                                 ^
    1 error generated.
    Makefile.WebCore.Target:70261: recipe for target '.obj/platform/network/MIMESniffing.o' failed
    make[3]: *** [.obj/platform/network/MIMESniffing.o] Error 1
    make[3]: *** Waiting for unfinished jobs....
    make[3]: Leaving directory '/private/tmp/qt520151023-46934-h365yy/qt-everywhere-opensource-src-5.5.1/qtwebkit/Source/WebCore'
    Makefile.WebCore:68: recipe for target 'sub-Target-pri-make_first-ordered' failed
    make[2]: *** [sub-Target-pri-make_first-ordered] Error 2
    make[2]: Leaving directory '/private/tmp/qt520151023-46934-h365yy/qt-everywhere-opensource-src-5.5.1/qtwebkit/Source/WebCore'
    Makefile:222: recipe for target 'sub-Source-WebCore-WebCore-pro-make_first-ordered' failed
    make[1]: *** [sub-Source-WebCore-WebCore-pro-make_first-ordered] Error 2
    make[1]: Leaving directory '/private/tmp/qt520151023-46934-h365yy/qt-everywhere-opensource-src-5.5.1/qtwebkit'
    Makefile:490: recipe for target 'module-qtwebkit-make_first' failed
    make: *** [module-qtwebkit-make_first] Error 2

    READ THIS: https://git.io/brew-troubleshooting

    These open issues may also help:
    qt5 5.5.0 errors in macdeployqt https://github.com/Homebrew/homebrew/issues/42191
    qt5: formula improvements & 5.5.1 version bump preparations https://github.com/Homebrew/homebrew/issues/44998
    qt5: not working properly when installed from within Travis CI https://github.com/Homebrew/homebrew/issues/44446
    qscintilla2 fails to build on 10.11: sip files generating qt5 API, but only qt4 installed https://github.com/Homebrew/homebrew/issues/41464
    qt5 (5.5.0) failed to build on OSX 10.10.4 https://github.com/Homebrew/homebrew/issues/42217
    Cannot build software that depends on Qt5 on CLT-only systems https://github.com/Homebrew/homebrew/issues/32431

Previously I installed Qt 5.3.2 with homebrew without a problem. I don't think I specified --with-developer though. I see that issue #42217 is about errors when building --with-docs. I don't know if that's related, but I'll reference it here anyways.

**Comment by dunn** [comment]
Could you post the full `brew gist-logs qt5`, please? Thanks!

**Comment by dagoodma** [comment]
https://gist.github.com/9a1def27b8b117ea5aad

**Comment by UniqMartin** [comment]
It looks like the Qt build process is picking up headers from your `llvm` installed via Homebrew. Are you building with `clang` provided by the `llvm` formula? If so, you might want to retry with the compiler provided by Xcode (maybe `brew unlink llvm` for the time being).

**Comment by dagoodma** [comment]
Hello. I unlinked the brewed llvm, removed it from my path, and checked that clang and clang++ were the Apple versions:

    $ clang++ --version
    Apple LLVM version 4.2 (clang-425.0.28) (based on LLVM 3.2svn)
    Target: x86_64-apple-darwin11.4.2
    Thread model: posix

I also checked `brew --env` to see that it is indeed using clang, but I'm sure if it's using the correct headers or not:

	$ brew --env
	HOMEBREW_CC: clang
	HOMEBREW_CXX: clang++
	MAKEFLAGS: -j2
	CMAKE_PREFIX_PATH: /usr/local
	CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
	CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
	PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.7
	ACLOCAL_PATH: /usr/local/share/aclocal
	PATH: /usr/local/Library/ENV/4.3:/usr/bin:/bin:/usr/sbin:/sbin

I am locked into this version of Apple clang for Mac OS X 10.7.5, since apple has stopped supporting my 2008 macbook. So upgrading Xcode past v4.6.3 is not an option for me =(.

When I tried updating Qt to 5.5.1 I saw the same error. This time brew is indeed using the Apple compiler at: `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang`. Here's the output:

	$ brew install qt5 --with-developer
	==> Downloading https://download.qt.io/official_releases/qt/5.5/5.5.1/single/qt-everywhere-opensource-src-5.5.1.tar.xz
	Already downloaded: /Library/Caches/Homebrew/qt5-5.5.1.tar.xz
	==> Downloading https://raw.githubusercontent.com/Homebrew/patches/2fcc1f8ec1df1c90785f4fa6632cebac68772fa9/qt5/el-capitan-2.diff
	Already downloaded: /Library/Caches/Homebrew/qt5--patch-b8f04efd047eeed7cfd15b029ece20b5fe3c0960b74f7a5cb98bd36475463227.diff==> Downloading https://gist.githubusercontent.com/UniqMartin/a54542d666be1983dc83/raw/f235dfb418c3d0d086c3baae520d538bae0b1c70/qtbug-47162.patch
	Already downloaded: /Library/Caches/Homebrew/qt5--patch-e31df5d0c5f8a9e738823299cb6ed5f5951314a28d4a4f9f021f423963038432.patch
	==> Patching
	patching file qtwebengine/src/3rdparty/chromium/chrome/installer/mac/third_party/bsdiff/goobsdiff.gyp
	patching file qtwebengine/src/3rdparty/chromium/content/content_browser.gypi
	patching file qtwebengine/src/3rdparty/chromium/net/net.gyp
	patching file qtwebengine/src/3rdparty/chromium/printing/printing.gyp
	patching file qtwebengine/src/3rdparty/chromium/sandbox/mac/sandbox_mac.gypi
	patching file qtwebengine/src/3rdparty/chromium/third_party/ffmpeg/ffmpeg.gyp
	patching file qtwebengine/src/3rdparty/chromium/third_party/libxml/libxml.gyp
	patching file qtwebengine/src/3rdparty/chromium/third_party/skia/gyp/zlib.gyp
	patching file qtwebengine/src/3rdparty/chromium/third_party/sqlite/sqlite.gyp
	patching file qtwebengine/src/3rdparty/chromium/tools/gyp/pylib/gyp/generator/xcode.py
	patch unexpectedly ends in middle of line
	Hunk #1 succeeded at 88 with fuzz 2.
	patching file qtbase/qmake/generators/makefile.cpp
	==> ./configure -prefix /usr/local/Cellar/qt5/5.5.1 -release -opensource -confirm-license -system-zlib -qt-libpng -qt-libjpeg -no-openssl -securetransport -nomake tests -no-
	==> make
	Last 15 lines from /Users/dagoodma/Library/Logs/Homebrew/qt5/02.make:
	inline _LIBCPP_INLINE_VISIBILITY const char* strchr(const char* __s, int __c) {return ::strchr(__s, __c);}
	                                             ^
	1 error generated.
	Makefile.WebCore.Target:70261: recipe for target '.obj/platform/network/MIMESniffing.o' failed
	make[3]: *** [.obj/platform/network/MIMESniffing.o] Error 1
	make[3]: *** Waiting for unfinished jobs....
	make[3]: Leaving directory '/private/tmp/qt520151024-30500-1vf0y2a/qt-everywhere-opensource-src-5.5.1/qtwebkit/Source/WebCore'
	Makefile.WebCore:68: recipe for target 'sub-Target-pri-make_first-ordered' failed
	make[2]: *** [sub-Target-pri-make_first-ordered] Error 2
	make[2]: Leaving directory '/private/tmp/qt520151024-30500-1vf0y2a/qt-everywhere-opensource-src-5.5.1/qtwebkit/Source/WebCore'
	Makefile:222: recipe for target 'sub-Source-WebCore-WebCore-pro-make_first-ordered' failed
	make[1]: *** [sub-Source-WebCore-WebCore-pro-make_first-ordered] Error 2
	make[1]: Leaving directory '/private/tmp/qt520151024-30500-1vf0y2a/qt-everywhere-opensource-src-5.5.1/qtwebkit'
	Makefile:490: recipe for target 'module-qtwebkit-make_first' failed
	make: *** [module-qtwebkit-make_first] Error 2

	READ THIS: https://git.io/brew-troubleshooting

	These open issues may also help:
	qt5 5.5.0 errors in macdeployqt https://github.com/Homebrew/homebrew/issues/42191
	qt5: formula improvements & 5.5.1 version bump preparations https://github.com/Homebrew/homebrew/issues/44998
	qt5: not working properly when installed from within Travis CI https://github.com/Homebrew/homebrew/issues/44446
	qscintilla2 fails to build on 10.11: sip files generating qt5 API, but only qt4 installed https://github.com/Homebrew/homebrew/issues/41464
	qt5 (5.5.0) failed to build on OSX 10.10.4 https://github.com/Homebrew/homebrew/issues/42217
	Cannot build software that depends on Qt5 on CLT-only systems https://github.com/Homebrew/homebrew/issues/32431

Here is the `brew gist-logs qt5`: https://gist.github.com/51364ca22a314dd3ee25

**Comment by UniqMartin** [comment]
Thanks for giving this a try! Unfortunately, the situation hasn't changed much and it's still picking up headers from your brewed `llvm`, as you can see at the end of the `02.make` log file:

```
platform/network/MIMESniffing.cpp:70:10: error: call to 'strchr' is ambiguous
    if (!strchr(type, '/')) {
         ^~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk/usr/include/string.h:
86:7: note: candidate function
char    *strchr(const char *, int);
         ^
/usr/local/Cellar/llvm/HEAD/bin/../include/c++/v1/cstring:105:46: note: candidate function
inline _LIBCPP_INLINE_VISIBILITY const char* strchr(const char* __s, int __c) {return ::strchr(__s, __c);}
                                             ^
1 error generated.
```

I'm a bit at a loss why this is happening. Maybe someone else has ideas how to figure that out. For the time being, I have two suggestions for you:

- If you can live without the `QtWebKit` module, try `brew install qt5 --without-webkit`. (You can also save a lot of compilation time if you disable `QtWebEngine` using `--without-webengine`. Software that requires these Qt modules will of course break.)
- If you really need the `QtWebKit` module, you can try reinstalling `qt5` after completely removing your brewed `llvm` with `brew remove llvm`. (Maybe make a backup of your `/usr/local/Cellar/llvm` before doing that.)

**Comment by dagoodma** [comment]
Hello Uniq,

I backed up llvm and removed it from brew as you mentioned. It turns out I had my xcode clang binary linked to the brewed llvm clang. I noticed this when I removed llvm. I ran `brew install qt5 --with-developer` again. This time it fails (gist: https://gist.github.com/a3dab497f03e672760fd ) here:

    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -c -Xclang -include-pch -Xclang .pch/QtCore/c++.pch -pipe -O3 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk -std=c++11 -stdlib=libc++ -mmacosx-version-min=10.7 -fvisibility=hidden -fvisibility-inlines-hidden -Wall -W -Werror -Wno-error=#warnings -Wno-error=deprecated-declarations -fPIC -DQT_NO_MTDEV -DQT_NO_LIBUDEV -DQT_NO_EVDEV -DQT_NO_TSLIB -DQT_NO_LIBINPUT -DQT_NO_USING_NAMESPACE -DQT_BUILD_CORE_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -DQT_NO_ICONV -DQT_CORE_LIB -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -I. -I../../include -I../../include/QtCore -I../../include/QtCore/5.5.1 -I../../include/QtCore/5.5.1/QtCore -Iglobal -I../3rdparty/pcre -I../3rdparty/harfbuzz/src -I../3rdparty/md5 -I../3rdparty/md4 -I../3rdparty/sha3 -I../3rdparty/forkfd -I.moc -I../../mkspecs/macx-clang -o .obj/forkfd_qt.o io/forkfd_qt.cpp
    In file included from io/forkfd_qt.cpp:80:
    io/../../3rdparty/forkfd/forkfd.c:150:5: error: expression result unused [-Werror,-Wunused-value]
        ffd_atomic_add_fetch(&header->busyCount, -1, FFD_ATOMIC_RELAXED);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    io/forkfd_qt.cpp:75:89: note: expanded from macro 'ffd_atomic_add_fetch'
    #define ffd_atomic_add_fetch(ptr,val,order) ((ptr)->FFD_CONCAT(fetchAndAdd, order)(val) + val)
                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
    In file included from io/forkfd_qt.cpp:80:
    io/../../3rdparty/forkfd/forkfd.c:244:5: error: expression result unused [-Werror,-Wunused-value]
        ffd_atomic_add_fetch(&header->busyCount, -1, FFD_ATOMIC_RELEASE);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    io/forkfd_qt.cpp:75:89: note: expanded from macro 'ffd_atomic_add_fetch'
    #define ffd_atomic_add_fetch(ptr,val,order) ((ptr)->FFD_CONCAT(fetchAndAdd, order)(val) + val)
                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
    2 errors generated.
    Makefile:45997: recipe for target '.obj/forkfd_qt.o' failed
    make[3]: *** [.obj/forkfd_qt.o] Error 1
    make[3]: *** Waiting for unfinished jobs....
    make[3]: Leaving directory '/private/tmp/qt520151024-86199-1c8nht1/qt-everywhere-opensource-src-5.5.1/qtbase/src/corelib'
    Makefile:166: recipe for target 'sub-corelib-make_first' failed
    make[2]: *** [sub-corelib-make_first] Error 2
    make[2]: Leaving directory '/private/tmp/qt520151024-86199-1c8nht1/qt-everywhere-opensource-src-5.5.1/qtbase/src'
    Makefile:45: recipe for target 'sub-src-make_first' failed
    make[1]: *** [sub-src-make_first] Error 2
    make[1]: Leaving directory '/private/tmp/qt520151024-86199-1c8nht1/qt-everywhere-opensource-src-5.5.1/qtbase'
    Makefile:72: recipe for target 'module-qtbase-make_first' failed
    make: *** [module-qtbase-make_first] Error 2
    /usr/local/Library/Homebrew/debrew.rb:11:in `raise'
    ==> Formula
    Path: /usr/local/Library/Formula/qt5.rb
    ==> Configuration
    HOMEBREW_VERSION: 0.9.5
    ORIGIN: https://github.com/Homebrew/homebrew.git
    HEAD: ffc24ff8ef42801875732e39cc2defd1d0a66318
    Last commit: 15 hours ago
    HOMEBREW_PREFIX: /usr/local
    HOMEBREW_REPOSITORY: /usr/local
    HOMEBREW_CELLAR: /usr/local/Cellar
    HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
    CPU: dual-core 64-bit penryn
    OS X: 10.7.5-i386
    Xcode: 4.6.3
    CLT: 4.6.0.0.1.1365549073
    LLVM-GCC: build 2336
    Clang: 4.2 build 425
    X11: 2.6.3 => /usr/X11
    System Ruby: 1.8.7-p358
    Perl: /usr/bin/perl
    Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.6_1/Frameworks/Python.framework/Versions/2.7/bin/python2.7
    Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.2.2/bin/ruby
    Java: 1.7.0_79
    ==> ENV
    HOMEBREW_CC: clang
    HOMEBREW_CXX: clang++
    MAKEFLAGS: -j2
    CMAKE_PREFIX_PATH: /usr/local
    CMAKE_INCLUDE_PATH: /usr/include/libxml2:/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
    CMAKE_LIBRARY_PATH: /System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
    PKG_CONFIG_PATH: /usr/local/opt/xz/lib/pkgconfig
    PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.7
    ACLOCAL_PATH: /usr/local/share/aclocal
    PATH: /usr/local/Library/ENV/4.3:/usr/local/opt/xz/bin:/usr/bin:/bin:/usr/sbin:/sbin

    Error: qt5 5.5.1 did not build
    Logs:
         /Users/dagoodma/Library/Logs/Homebrew/qt5/01.configure
         /Users/dagoodma/Library/Logs/Homebrew/qt5/02.make
         /Users/dagoodma/Library/Logs/Homebrew/qt5/02.make.cc

    These open issues may also help:
    qt5 5.5.0 errors in macdeployqt https://github.com/Homebrew/homebrew/issues/42191
    qt5: formula improvements & 5.5.1 version bump preparations https://github.com/Homebrew/homebrew/issues/44998
    qt5: not working properly when installed from within Travis CI https://github.com/Homebrew/homebrew/issues/44446
    qscintilla2 fails to build on 10.11: sip files generating qt5 API, but only qt4 installed https://github.com/Homebrew/homebrew/issues/41464
    qt5 (5.5.0) failed to build on OSX 10.10.4 https://github.com/Homebrew/homebrew/issues/42217
    Cannot build software that depends on Qt5 on CLT-only systems https://github.com/Homebrew/homebrew/issues/32431


Currently, I don't need webkit or webengine. I only need the following components:

    QtConcurrent.framework
    QtCore.framework
    QtDBus.framework
    QtGui.framework
    QtLocation.framework
    QtNetwork.framework
    QtOpenGL.framework
    QtPositioning.framework
    QtPrintSupport.framework
    QtQml.framework
    QtQuick.framework
    QtQuickWidgets.framework
    QtSql.framework
    QtSvg.framework
    QtWidgets.framework
    QtXml.framework

I installed without webkit or webengine as you said, and I see the exact same error. It seems to be occurring in forkfd.

**Comment by UniqMartin** [comment]
> I backed up llvm and removed it from brew as you mentioned. It turns out I had my xcode clang binary linked to the brewed llvm clang. I noticed this when I removed llvm.

Good, at least we know now where those include paths were coming from.

> I ran brew install `qt5 --with-developer` again.

Please don't use `--with-developer`. It's not necessary if you are developing an application that uses the Qt framework. Among other things, it turns warns warnings into errors and that's almost guaranteed to break the Qt build. (That's what happened to your build now.)

> Currently, I don't need webkit or webengine. I only need the following components: […]

Given your requirements, I'd strongly suggest to install Qt 5 with `brew --with-d-bus --without-webengine --without-webkit`.

**Comment by mikemcquaid** [comment]
> Please don't use --with-developer. It's not necessary if you are developing an application that uses the Qt framework. Among other things, it turns warnings into errors and that's almost guaranteed to break the Qt build. (That's what happened in your most recent attempt.)

We should probably remove this option.

**Comment by UniqMartin** [comment]
> We should probably remove this option.

I agree, probably too misleading and not very useful in the context of Homebrew. Should I sneak that into the `qt5` revision bump that will hopefully resolve the remaining outstanding issues? (Yes, I'm still working on that. Update coming soon.)

**Comment by mikemcquaid** [comment]
> I agree, probably too misleading and not very useful in the context of Homebrew. Should I sneak that into the upcoming qt5 revision bump that will hopefully resolve the remaining outstanding issues? (Yes, I'm still working on that. Update coming soon.)

:+1:

**Comment by dagoodma** [comment]
Thanks for the tips. I recompiled with `brew install qt5 --with-d-bus --without-webengine --without-webkit`, and it worked for a while. Then, brew reported an error when making xmlpatterns, but I couldn't figure out what the error was or where it occurred. I re-ran with `install -vd`, then went to the shell and executed the "Failing" make command. Seemed to run fine, and exit status $?=0. Here's the gist: 
https://gist.github.com/anonymous/8cea6d0f0fd852bfc544, and output (none) of running the make command:

    [dagoodma][macbook][/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtxmlpatterns/tools/xmlpatterns]
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -headerpad_max_install_names -Wl,-dead_strip -Wl,-syslibroot,/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk -stdlib=libc++ -mmacosx-version-min=10.7 -o ../../bin/xmlpatterns .obj/main.o .obj/qapplicationargument.o .obj/qapplicationargumentparser.o   -F/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtxmlpatterns/lib -F/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtbase/lib -L/usr/local/opt/d-bus/lib -framework QtXmlPatterns -framework QtNetwork -framework QtCore -framework DiskArbitration -framework IOKit -ldbus-1

I continued building after ignoring the error, and then I saw another error about missing -lpoly2tri, the delanauy triangulation library:

	/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -headerpad_max_install_names -Wl,-syslibroot,/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk -stdlib=libc++ -mmacosx-version-min=10.7 -single_module -dynamiclib -o libdeclarative_location.dylib .obj/location.o .obj/qdeclarativegeomapitemview.o .obj/qdeclarativegeoserviceprovider.o .obj/qdeclarativegeocodemodel.o .obj/qdeclarativegeoroutemodel.o .obj/qdeclarativegeoroute.o .obj/qdeclarativegeoroutesegment.o .obj/qdeclarativegeomaneuver.o .obj/qdeclarativegeomap.o .obj/qdeclarativegeomaptype.o .obj/qdeclarativegeomapitembase.o .obj/qdeclarativegeomapquickitem.o .obj/qdeclarativecirclemapitem.o .obj/qdeclarativerectanglemapitem.o .obj/qdeclarativepolygonmapitem.o .obj/qdeclarativepolylinemapitem.o .obj/qdeclarativeroutemapitem.o .obj/qgeomapitemgeometry.o .obj/qdeclarativegeomapcopyrightsnotice.o .obj/qdeclarativegeomapgesturearea.o .obj/error_messages.o .obj/locationvaluetypehelper.o .obj/qdeclarativeplacecontentmodel.o .obj/qdeclarativesupportedcategoriesmodel.o .obj/qdeclarativesearchsuggestionmodel.o .obj/qdeclarativesearchresultmodel.o .obj/qdeclarativereviewmodel.o .obj/qdeclarativeplaceimagemodel.o .obj/qdeclarativeplaceeditorialmodel.o .obj/qdeclarativecontactdetail.o .obj/qdeclarativecategory.o .obj/qdeclarativeplace.o .obj/qdeclarativeplaceattribute.o .obj/qdeclarativeplaceicon.o .obj/qdeclarativeplaceuser.o .obj/qdeclarativeratings.o .obj/qdeclarativesupplier.o .obj/qdeclarativesearchmodelbase.o .obj/moc_qdeclarativecirclemapitem_p.o .obj/moc_qdeclarativerectanglemapitem_p.o .obj/moc_qdeclarativepolygonmapitem_p.o .obj/moc_qdeclarativepolylinemapitem_p.o .obj/moc_qdeclarativeroutemapitem_p.o .obj/moc_qdeclarativegeomapcopyrightsnotice_p.o .obj/moc_qdeclarativeplacecontentmodel.o .obj/moc_qdeclarativesupportedcategoriesmodel_p.o .obj/moc_qdeclarativesearchsuggestionmodel_p.o .obj/moc_qdeclarativesearchresultmodel_p.o .obj/moc_qdeclarativereviewmodel_p.o .obj/moc_qdeclarativeplaceimagemodel_p.o .obj/moc_qdeclarativeplaceeditorialmodel.o .obj/moc_qdeclarativecontactdetail_p.o .obj/moc_qdeclarativecategory_p.o .obj/moc_qdeclarativeplace_p.o .obj/moc_qdeclarativeplaceattribute_p.o .obj/moc_qdeclarativeplaceicon_p.o .obj/moc_qdeclarativeplaceuser_p.o .obj/moc_qdeclarativeratings_p.o .obj/moc_qdeclarativesupplier_p.o .obj/moc_qdeclarativesearchmodelbase.o  -F/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtdeclarative/lib -F/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtlocation/lib -F/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtbase/lib -L/usr/local/opt/d-bus/lib -framework QtLocation -framework QtPositioning -framework QtCore -framework DiskArbitration -framework IOKit -framework QtQuick -framework QtGui -framework QtQml -framework QtNetwork -ldbus-1 -framework OpenGL -framework AGL -L/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtlocation/lib -lpoly2tri -L/usr/local/opt/d-bus/lib -ldbus-1
	ld: library not found for -lpoly2tri
	clang: error: linker command failed with exit code 1 (use -v to see invocation)
	Makefile:295: recipe for target '../../../qml/QtLocation/libdeclarative_location.dylib' failed
	make[4]: *** [../../../qml/QtLocation/libdeclarative_location.dylib] Error 1
	make[4]: Leaving directory '/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtlocation/src/imports/location'
	Makefile:80: recipe for target 'sub-location-install_subtargets' failed
	make[3]: *** [sub-location-install_subtargets] Error 2
	make[3]: Leaving directory '/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtlocation/src/imports'
	Makefile:135: recipe for target 'sub-imports-install_subtargets' failed
	make[2]: *** [sub-imports-install_subtargets] Error 2
	make[2]: Leaving directory '/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtlocation/src'
	Makefile:56: recipe for target 'sub-src-install_subtargets' failed
	make[1]: *** [sub-src-install_subtargets] Error 2
	make[1]: Leaving directory '/private/tmp/qt520151025-33370-10b5rms/qt-everywhere-opensource-src-5.5.1/qtlocation'
	Makefile:363: recipe for target 'module-qtlocation-install_subtargets' failed
	make: *** [module-qtlocation-install_subtargets] Error 2
	/usr/local/Library/Homebrew/debrew.rb:11:in `raise'
	BuildError: Failed executing: make install

What now? Shouldn't poly2tri (a google library) be contained in Qt or listed as a dependency?

**Comment by UniqMartin** [comment]
The library `poly2tri` is distributed together with QtLocation and should have been already built by the time the error occurs. Why that doesn't happen in your case or why it fails to find the library that was just built, I don't know. (And I failed to find any indication of this in your log files.)

As for the error that does show up in the log: The missing `__builtin_bswap16` function doesn't exist in your version of Clang, but the offending code seems to believe it is there. Not sure what can be done about this.

**Comment by dagoodma** [comment]
I think the reason I initially installed brewed llvm was due to the missing `__builtin_bswap16` function in my outdated Apple clang. Maybe QtLocation failed because of the missing function, and then when I ignored and continued building missing QtLocation caused poly2tri not to be found.

**Comment by DomT4** [comment]
Might be worth forcing GCC-5 and seeing if it will build with that, perhaps.

**Comment by uliano** [comment]
gcc5 doesn't work
```
g++-5: error: unrecognized command line option '-stdlib=libc++'
```
here is the new gist

https://gist.github.com/05276192f61f0f8ac843

@UniqMartin sorry for having posted this elsewhere, I still understand little of github "mechanics" and interpreted the "Cloesd" above as "you shall not post here"

**Comment by UniqMartin** [comment]
> sorry for having posted this elsewhere, I still understand little of github "mechanics" and interpreted the "Cloesd" above as "you shall not post here"

No worries and thanks for moving it here! The “Closed” you see above your comment here is the status of the issue left of it. The status of this issue is at the very top under the title and still “Open”.

Would you mind editing `qt5.rb` and trying again to build with GCC 5? If not, `brew edit qt5` should open the formula file in your default editor on the command line. You can also open `/usr/local/Library/Formula/qt5.rb` directly. Add this line after line 75:

```
      -platform macx-g++
```

Afterwards this piece of code should look something like this:

```
    args = %W[
      -platform macx-g++
      -prefix #{prefix}
      -release
```

(Note that this is only a temporary hack.)

**Comment by uliano** [comment]
after editing qt5.rb 

```
bash-3.2$ brew upgrade qt5
…
==> ./configure -platform macx-g++ -prefix /usr/local/Cellar/qt5/5.5.1_1 -release -opensource -confirm-license -system-zlib -qt-libpng -qt-libjpeg -no-openssl -securet
...
```
(without forcing gcc-5)

and the new gist

https://gist.github.com/27dcf7fdbf80679c90d3

**Comment by UniqMartin** [comment]
In addition to editing the file you still have to use `--cc=gcc-5` as in `brew install qt5 --cc=gcc-5`. Otherwise, the Qt build system won't see the GCC 5 built by Homebrew and instead use the `gcc` and `g++` binaries from Xcode (and this is not what we want).

**Comment by uliano** [comment]
```
bash-3.2$ brew upgrade qt5 --cc=gcc-5
```
maybe upgrade and install behave differently as it seems to me that the same apple compiler as before is being used

```
.../Applications/Xcode.app/Contents/Developer/usr/bin/g++ -c -include...
```

https://gist.github.com/6c353c2a687a59f09347

**Comment by UniqMartin** [comment]
You are right. I think I misunderstood the meaning of `-platform macx-g++`. It's there to support the GCC versions provided by Apple, that are even older than the Clang you have as part of Xcode. As far as I can tell there's no support for building with other compilers on OS X, especially not some arbitrary GCC. And because Qt doesn't seem to care about OS X 10.7 (understandably though, as 10.9 is the oldest still supported OS X release) you are now on your own.

The best suggestions I can give at this point are:

- Hack the Qt build system to make it work with any GCC on OS X. This is probably non-trivial.
- Build on a newer system and then copy from there. Qt 5 has very few external dependencies and is built with a deployment target of 10.7, so it will probably continue to work even if built on, e.g., 10.9 and then copied back to a 10.7 system.
- Revert to an older Qt 5 version that worked well with 10.7.

**Comment by DomT4** [comment]
Closing in favour of https://github.com/Homebrew/homebrew/pull/45940.

**Comment by DomT4** [comment]
In terms of your specific problem, I'd revert back to a version of Qt 5 that built for you. After that you're stuck there I'm afraid. If you need it for non-brew purposes you *may* find the upstream binaries work well on older systems.
